lib/timeline/timeline.dart

```diff
@@ -46,9 +46,29 @@ class TimelineScreen extends Screen {
   PButton resumeButton;
 
   @override
-  void createContent(Framework framework, CoreElement mainDiv) {
+  void createContent(Framework framework, CoreElement mainDiv) async {
     FrameDetailsUI frameDetailsUI;
 
+    final Isolate isolate = await serviceInfo.service
+        .getIsolate(serviceInfo.isolateManager.selectedIsolate.id);
+
+    bool isProfileMode;
+
+    // Determine whether we are in profile mode. If evaluate successfully returns
+    // an InstanceRef, we are in debug mode. If evaluate throws an error, we are
+    // in profile mode. Expression '1+1' is arbitrary.
+    // TODO(kenzieschmoll): The VMService should expose the build mode (debug/profile).
+    try {
+      final dynamic evaluate = await serviceInfo.service.evaluate(
+        serviceInfo.isolateManager.selectedIsolate.id,
+        isolate.rootLib.id,
+        '1+1',
+      );
+      isProfileMode = !(evaluate is InstanceRef);
+    } catch (e) {
+      isProfileMode = true;
```

Comment:
What exception is thrown? Instead of catching all exceptions verify we got the expected one.